### PR TITLE
Add documentation for update_query method

### PR DESCRIPTION
**Pull request by mosquito**
Documentation and some notes for `update_query` method.

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -463,6 +463,50 @@ section generates a new *URL* instance.
       >>> URL('http://example.com/path?a=b&b=1').with_query([('b', '2')])
       URL('http://example.com/path?b=2')
 
+.. method:: URL.update_query(query)
+            URL.update_query(**kwargs)
+
+   Returns a new URL with *query* part updated.
```
Comment (asvetlov) [inline docs/api.rst]: The difference with `with_query` should be highlighted here.
Doc for `with_query` also should mention `update_query` with words like "for replacing only particular query keys use update_query"

**Comment by mosquito** [comment]
@asvetlov fixed.

**Comment by mosquito** [comment]
@asvetlov can you merge it?

**Comment by asvetlov** [comment]
Thanks

**Comment by asvetlov** [comment]
Polished by 3f76f2f

**Pull request merged**